python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py

```diff
@@ -0,0 +1,299 @@
+            )
+
+    async def open_white_light_shutter(self):
+        await self.white_light_source.cmd_openShutter.start(
+            timeout=self.timeout_open_shutter
+        )
+
+    async def switch_lamp_on(self):
+        self.white_light_source.evt_lampState.flush()
+
+        await self.white_light_source.cmd_turnLampOn.set_start(
+            power=self.whitelight_power, timeout=self.timeout_lamp_warm_up
+        )
+
+    async def wait_for_lamp_to_warm_up(self):
+        lamp_state = await self.white_light_source.evt_lampState.aget(
+            timeout=self.timeout_lamp_warm_up
+        )  # gets the most recent data
+        self.log.info(
+            f"Lamp state: {ATWhiteLight.LampBasicState(lamp_state.basicState)!r}."
+        )
+
+        # Now enters the loop with the stop condition being that the lamp is ON
+
+        while lamp_state.basicState != ATWhiteLight.LampBasicState.ON:
+            try:
+                lamp_state = await self.white_light_source.evt_lampState.next(
+                    flush=False, timeout=self.timeout_lamp_warm_up
+                )
+                self.log.info(
+                    f"Lamp state: {ATWhiteLight.LampBasicState(lamp_state.basicState)!r}."
+                )
+            except asyncio.TimeoutError:
+                raise RuntimeError(
+                    f"White Light Lamp failed to turn on after {self.timeout_lamp_warm_up} s."
+                )
+
+    async def configure_atmonochromator(self):
+        await self.monochromator.cmd_selectGrating.set_start(
+            gratingType=self.grating_type, timeout=self.cmd_timeout
+        )
+
+        await self.monochromator.cmd_changeWavelength.set_start(
+            wavelength=self.wavelength, timeout=self.cmd_timeout
+        )
+
+        await self.monochromator.cmd_changeSlitWidth.set_start(
+            slit=1, slitWidth=self.entrance_slit_width, timeout=self.cmd_timeout
+        )
+
+        await self.monochromator.cmd_changeSlitWidth.set_start(
+            slit=2, slitWidth=self.exit_slit_width, timeout=self.cmd_timeout
+        )
+
+        params = await self.get_monochromator_parameters()
+
+        self.log.info(
+            f"ATMonochromator grating is {params[0]}, \n"
+            f"wavelength is {params[1]}, \n"
+            f"with entry slit width {params[2]}, \n"
+            f"and exit slit width {params[3]}"
+        )
+
+    async def get_monochromator_parameters(self):
+        return await asyncio.gather(
+            self.monochromator.evt_selectedGrating.aget(timeout=self.cmd_timeout),
+            self.monochromator.evt_wavelength.aget(timeout=self.cmd_timeout),
+            self.monochromator.evt_entrySlitWidth.aget(timeout=self.cmd_timeout),
+            self.monochromator.evt_exitSlitWidth.aget(timeout=self.cmd_timeout),
+        )
+
+    async def assert_components_enabled(self):
+        """Check if ATWhiteLight and ATMonochromator are ENABLED"""
```

Comment:
Please, add a `Raises` session to the docstring and explain that it raises a `RuntimeError`, e.g.;

```
Raises
------
RuntimeError:
  If ATWhiteLight or ATMonochromator are not in ENABLED state.
```